src/Suave/Compression.fs

```diff
@@ -75,51 +75,63 @@ module Compression =
-    use new_fs = new FileStream(path, FileMode.CreateNew, FileAccess.Write, FileShare.Write)
+    use newFileStream = new FileStream(path, FileMode.CreateNew, FileAccess.Write, FileShare.Write)
     match encoding with
     | GZIP ->
-      use gzip = new GZipStream(new_fs, CompressionMode.Compress)
-      do! lift_task (fs.CopyToAsync gzip)
+      use gzip = new GZipStream(newFileStream, CompressionMode.Compress)
+      do! liftTask (fs.CopyToAsync gzip)
       gzip.Close()
     | Deflate ->
-      use deflate = new DeflateStream(new_fs, CompressionMode.Compress)
-      do! lift_task (fs.CopyToAsync deflate)
+      use deflate = new DeflateStream(newFileStream, CompressionMode.Compress)
+      do! liftTask (fs.CopyToAsync deflate)
       deflate.Close()
     | _ ->
       return failwith "invalid case."
-    new_fs.Close()
+    newFileStream.Close()
   }
 
-  let compress_file n (stream : Stream) compression_folder = socket {
-    let temp_file_name = Path.GetRandomFileName()
-    if not (Directory.Exists compression_folder) then Directory.CreateDirectory compression_folder |> ignore
-    let new_path = Path.Combine(compression_folder,temp_file_name)
+  let compressFile n (stream : Stream) compressionFolder = socket {
+    let tempFileName = Path.GetRandomFileName()
+    if not (Directory.Exists compressionFolder) then Directory.CreateDirectory compressionFolder |> ignore
+    let new_path = Path.Combine(compressionFolder,tempFileName)
     do! compress n new_path stream
     stream.Dispose()
     return new_path
   }
 
-  let transform_x (key : string) (get_data : string -> Stream) (get_last : string -> DateTime) compression compression_folder ({ request = q } as ctx) connection =
+  let transformStream (key : string) (getData : string -> Stream) (getLast : string -> DateTime) compression compressionFolder ({ request = q } as ctx) connection =
     socket {
-      let stream = get_data key
+      let stream = getData key
       if compression && stream.Length > int64(MIN_BYTES_TO_COMPRESS) && stream.Length < int64(MAX_BYTES_TO_COMPRESS) then
-        let enconding = parse_encoder q
+        let enconding = parseEncoder q
         match enconding with
         | Some (n) ->
-          do! async_writeln connection (String.Concat [| "Content-Encoding: "; n.ToString() |])
-          if Globals.compressed_files_map.ContainsKey key then
-            let last_modified = get_last key
-            let cmpr_info = new FileInfo(Globals.compressed_files_map.[key])
-            if last_modified > cmpr_info.CreationTime then
-              let! new_path =  compress_file n stream compression_folder
-              Globals.compressed_files_map.[key] <- new_path
+          do! asyncWriteLn connection (String.Concat [| "Content-Encoding: "; n.ToString() |])
+          if Globals.compressedFilesMap.ContainsKey key then
+            let lastModified = getLast key
+            let cmpr_info = new FileInfo(Globals.compressedFilesMap.[key])
```

Comment:
It's just an internal table so doesn't matter too much. Perhaps it should be changed from a ConcurrentDictionary to System.Runtime.Caching.ObjectCache or the like.